Improve releasing to rancher/charts automation script

When releasing Turtles v0.26.1 (un-rc'ing) version, `rancher-charts` repo PR opening automation opened a PR: https://github.com/rancher/charts/pull/7367, which failed [CI](https://github.com/rancher/charts/actions/runs/24889904141/job/72880602294?pr=7367) validation, logs:

```
[FAIL] rancher-turtles generated git changes
[PASS] rancher-webhook
[PASS] remotedialer-proxy
[PASS] system-upgrade-controller
[PASS] ui-plugin-operator

========================================
[INFO] Package CI Summary
========================================
Total packages:   44
Passed:           37
Skipped:          6
Failed:           1
========================================

  - rancher-turtles: generated git changes
[FAIL] Failed packages:
    M packages/rancher-turtles/generated-changes/patch/Chart.yaml.patch

##[debug]{"message":"command terminated with non-zero exit code: error executing command [sh -e /__w/_temp/197eee60-3fdc-11f1-8bad-395caed0309f.sh], exit code 1","details":{"causes":[{"reason":"ExitCode","message":"1"}]}}
##[debug]execPodStep failed: "command terminated with non-zero exit code: error executing command [sh -e /__w/_temp/197eee60-3fdc-11f1-8bad-395caed0309f.sh], exit code 1"
Error: Error: failed to run script step: command terminated with non-zero exit code: error executing command [sh -e /__w/_temp/197eee60-3fdc-11f1-8bad-395caed0309f.sh], exit code 1
Error: Process completed with exit code 1.
Error: Executing the custom container implementation failed. Please contact your self-hosted runner administrator.
```
That is because https://github.com/rancher/charts/pull/7260 introduced a new CI validation which checks that
the latest version of any chart in `package/ `folder must always be reproducible and doesn't generate git changes. Manual workaround to fix the failing CI was to run the following commands and push the local changes to the branch:

```
PACKAGE=rancher-turtles make prepare
PACKAGE=rancher-turtles make icon
PACKAGE=rancher-turtles make patch 
PACKAGE=rancher-turtles make clean
PACKAGE=rancher-turtles make charts
```

suggested by @nicholasSUSE in slack [thread](https://suse.slack.com/archives/C09JDMPE37D/p1777035891834899).

We need to check our [automation script](https://github.com/rancher/turtles/blob/main/.github/scripts/release-against-charts.sh#L142-L153) and make necessary adjustments to perform the above 5 commands and commit each one separately.

--- a/.github/scripts/release-against-charts.sh
+++ b/.github/scripts/release-against-charts.sh
@@ -142,9 +142,15 @@ sed -i "s/${PREV_CHART_VERSION}/${NEW_CHART_VERSION}/g" ./packages/rancher-turtl
 git add packages/rancher-turtles
 git commit -m "$COMMIT_MSG"
 
+# Fully regenerate chart artifacts
+make pull-scripts
+PACKAGE=rancher-turtles make prepare
+PACKAGE=rancher-turtles make icon
+PACKAGE=rancher-turtles make patch
+PACKAGE=rancher-turtles make clean
 PACKAGE=rancher-turtles make charts
-git add ./assets/rancher-turtles ./charts/rancher-turtles index.yaml
-git commit -m "make charts"
+git add .
+git commit -m "PACKAGE=rancher-turtles make prepare/icon/patch/clean/charts"
 
 # Prepends to list
 yq --inplace ".rancher-turtles = [\"${NEW_CHART_VERSION}+up${NEW_TURTLES_VERSION_SHORT}\"] + .rancher-turtles" release.yaml

